lib/split-home-site-urls/login.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Automattic\VIP\Split_Home_Site_URLs;
+
+/**
+ * Rewrite any admin URLs to use the site URL, not the home URL
+ */
+function rewrite_admin_url_to_site_url( $redirect_to ) {
+	// Nothing to correct if it's not a wp-admin link
+	if ( empty( $redirect_to ) || false === stripos( $redirect_to, 'wp-admin' ) ) {
+		return $redirect_to;
+	}
+
+	// Grab the path relative to wp-admin
+	preg_match( '#/wp-admin(/)?(.+?)$#i', $redirect_to, $admin_paths );
```

Comment:
Another non-regex (maybe faster and simpler) approach would be to `explode()` the string on `/wp-admin` with a limit of 1.